src/library/masq.js

```diff
@@ -317,17 +317,19 @@ class Masq {
 
         if (msg === 'requestWriteAccess') {
           const userAppKey = Buffer.from(json.key, 'hex')
-          this.appsDBs[dbName].authorize(userAppKey, (err) => {
-            if (err) throw err
-            sendWriteAccessGranted(peer)
-            this.sw.close()
-            return resolve()
-          })
+          try {
+            await this.appsDBs[dbName].authorizeAsync(userAppKey)
+          } catch (err) {
```

Comment:
not necessary to catch right ?

Reply:
or if we need to catch we may close the connection

Reply:
Yes it is not necessary but we should not rethrow the same error, we should specify an error type with a MasqError

Reply:
Ok but maybe we need to close the connecttion at least ?

Reply:
I guess we do... but I think there is at least one other error case in which we do not close the connection, at the execution of _checkProfile, if it throws we should close the connection and rethrow ?